File: Java/프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    
    private boolean isPrime(int x) {
        if (x < 2) return false;
        
        for (int i = 2; i * i <= x; i++) {
            if (x % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    public int solution(int[] nums) {
        int answer = 0;
        int n = nums.length;
        
        for (int i = 0; i < n; i++){
            for (int j = i + 1; j < n; j++) {
                for (int k = j + 1; k < n; k++) {
                    int sum = nums[i] + nums[j] + nums[k];
                    if (isPrime(sum)) {
                        answer++;
                    }
                }
            }
        }
        return answer;
    }
}

File: Python3/프로그래머스/1/12977. 소수 만들기/소수 만들기.py
from itertools import combinations

def is_prime(x):
    if x < 2:
        return False
    for i in range(2, int(x ** 0.5) +1):
        if x % i == 0:
            return False
    return True

def solution(nums):
    answer = 0
    for a, b, c in combinations(nums, 3):
        if is_prime(a + b + c):
            answer += 1
    return answer